lib/src/request.dart

```diff
@@ -9,50 +10,49 @@ import 'package:api_utils/src/api_response.dart';
 
 @visibleForTesting
 void setHttpClientForTesting(http.Client client) {
-  assert(client != null);
   _client = client;
 }
 
 http.Client _client = http.Client();
 
-typedef FromJson<T> = T Function(Map<String, dynamic>);
+typedef FromJson<T> = T Function(Map<String, dynamic>?);
 
 /// Make a GET request with a json object response
 Future<ApiResponse<T>> get<T>(
-    {@required String url,
-    @required FromJson<T> fromJson,
+    {required String url,
+    required FromJson<T> fromJson,
     bool useFromJsonOnFailure = false,
-    Map<String, String> headers}) async {
+    required Map<String, String> headers}) async {
```

Comment:
Maybe we should default the headers to some sensible default. This way it can still be non null, but the users of the library only need to set it if needed.

Maybe we default to Content Type application/json.